src/NUnitFramework/framework/Diagnostics/ProgressTraceListener.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using System;
+using System.Diagnostics;
+#if !NET6_0_OR_GREATER
+using System.Security.Permissions;
+#endif
+
+namespace NUnit.Framework.Diagnostics
+{
+    /// <summary><para>
+    /// The ProgressTraceListener class allows directing tracing or
+    /// debugging output to <see cref="TestContext.Progress"/>.
+    /// </para><para>
+    /// To activate, place the following snippet into the one-time
+    /// set-up method of either a test's fixture or the set-up
+    /// fixture of a project:
+    /// <c>
+    /// System.Trace.Listeners.Add(new ProgressTraceListener());
+    /// </c>
+    /// </para><para>
+    /// Make sure to only add a listener once, e.g.:
+    /// <c>
+    /// if (!System.Trace.Listeners.OfType&lt;ProgressTraceListener&gt;().Any())
+    ///     System.Trace.Listeners.Add(new ProgressTraceListener());
+    /// </c>
+    /// </para><para>
+    /// Or again remove in the one-time tear-down, e.g.:
+    /// <c>
+    /// System.Trace.Listeners.Remove(_progressTraceListener);
+    /// </c>
+    /// </para></summary>
+    /// <remarks><para>
+    /// Although named "Trace", <see cref="TextWriterTraceListener"/>
+    /// "directs tracing or debugging output".
+    /// </para><para>
+    /// This listener is provided by NUnit (i.e. the origin of
+    /// <see cref="TestContext.Progress"/>) same as the
+    /// <see cref="ConsoleTraceListener"/> is provided by .NET
+    /// (the origin of <see cref="Console"/>).
+    /// </para></remarks>
+    // Implemented following the ConsoleTraceListener implementation.
+#if !NET6_0_OR_GREATER
+    [HostProtection(Synchronization = true)]
```

Comment:
@maettu-this I'm curious, is this needed? I haven't used this myself, but the documentation I can find indicates this could be [CAS-related](https://learn.microsoft.com/en-us/dotnet/api/system.security.permissions.hostprotectionattribute?view=net-8.0) which is something I think we tried to move away from in NUnit 4

Reply:
Even the .NET Framework [documentation](https://learn.microsoft.com/en-us/dotnet/api/system.security.permissions.hostprotectionattribute?view=netframework-4.6.2#remarks) says it has been deprecated. 
I do see `HostProtection` in the `ConsoleTraceListener`  .NET Framework reference [source](https://github.com/microsoft/referencesource/blob/master/System/compmod/system/diagnostics/ConsoleTraceListener.cs)

Reply:
@stevenaw and @manfred-brands, for this very reason I added the...

> // Implemented following the ConsoleTraceListener implementation.

...comment at line 42. I have neither used this attribute myself, but I am pretty sure the .NET framework guys have (or at least had) reasons for this.

Reply:
We can leave it in for now.